src/NHibernate.Test/NHSpecificTest/NH3530/Fixture.cs

```diff
@@ -0,0 +1,183 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NHibernate.Cfg;
+using NHibernate.Cfg.MappingSchema;
+using NHibernate.Loader;
+using NUnit.Framework;
+using Environment = NHibernate.Cfg.Environment;
+
+namespace NHibernate.Test.NHSpecificTest.NH3530
+{
+	//NH-3530 (GH-1316) 
+	[TestFixture]
+	public class DynamicBatchFetchStyleFixture : TestCaseMappingByCode
+	{
+		private readonly List<object> _ids = new List<object>();
+
+		protected override void Configure(Configuration configuration)
+		{
+			base.Configure(configuration);
+			configuration.SetProperty(Environment.BatchFetchStyle, BatchFetchStyle.Dynamic.ToString());
+		}
+
+		[Test]
+		public void CanLoadEntity()
+		{
+			PrepareEntities(2);
+
+			using (var session = OpenSession())
+			{
+				var proxy = session.Load<Entity>(_ids[0]);
+				var result = session.Get<Entity>(_ids[1]);
+
+				Assert.That(result.Name, Is.Not.Null);
+
+				var childrenCount = result.Children.Count;
+				//Assert.That(NHibernateUtil.IsInitialized(proxy), Is.True);
```

Comment:
This entity is already loaded by batch loading. But proxy doesn't know about it - NHibernateUtil.IsInitialized(proxy) returns false. Looks like an issue to me.